src/Psalm/Internal/Type/Comparator/ScalarTypeComparator.php

```diff
@@ -443,10 +443,12 @@ public static function isContainedBy(
                     ($max_bound === null || $max_bound >= $input_type_part->value);
             }
 
-            //any int can't be pushed inside a range without coercion
-            if ($atomic_comparison_result) {
-                $atomic_comparison_result->type_coerced = true;
-                $atomic_comparison_result->type_coerced_from_scalar = true;
+            //any int can't be pushed inside a range without coercion (unless the range is from min to max)
+            if ($container_type_part->min_bound !== null && $container_type_part->max_bound !== null) {
```

Comment:
Shouldn't that be OR instead of AND? 'If the range has any non-max/non-min bounds specified TInt won't fit'

Reply:
wow, I'm starting to make big mistakes. I have my head full of the stuff, I'll get back to it later. Right now I have build failures because `int<min, 1> + 1` is translated to `int`

Reply:
It does seem that those failures were resolved by this change though.

Reply:
You're right. Must be something to do with coercion being allowed in array access or something like that... It's good, it means I'll be able to put arithmetics in a separated chunk of changes